### PR TITLE
Add Systemd services for Podman and FCOS integration

### PR DESCRIPTION
**Pull request by eriksjolund**
* Add the Systemd unit files scout-*.services, so that
  Scout and Mongodb can be run as Systemd services with
  Podman.

* Add the file scout.fcc from which an Igntion file can
  be generated, so that Scout can be run on Fedora CoreOS

* Add the file var_lib_systemd_linger_core that is
  used in scout.fcc. It just acts as an empty file.
  The future plan is to remove the file and instead
  define an empty file in scout.fcc instead.

* Add the file README.md with documentation

Signed-off-by: Erik Sjölund <erik.sjolund@scilifelab.se>

This PR adds a functionality or fixes a bug.

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Ok I've been reading about podman and systemd so now I can take the review of this one!

**Comment by eriksjolund** [comment]
Maybe a GitHub Actions workflow should be added too?

A first test to run Scout with Systemd in a GitHub Actions workflow: 

```
git checkout -b systemd_services_podman_v3_github_actions
git add .github/workflows/test_systemd_user_service.yml
```

The workflow file is here:
https://github.com/eriksjolund/scout/blob/systemd_services_podman_v3_github_actions/.github/workflows/test_systemd_user_service.yml

The GitHub Actions workflow was able to run 
```
curl http://localhost:5000
``` 
For details see:

https://github.com/eriksjolund/scout/runs/1758291500?check_suite_focus=true

(the output of the curl command can be seen there)

The `sleep 30` is a bit of a hack. It was added to wait for the loading of the database.

**Comment by northwestwitch** [comment]
> Maybe a GitHub Actions workflow should be added too?

Yes, why not! I'll try to review this between today and tomorrow. Good job!

**Review by northwestwitch — CHANGES_REQUESTED** [review]
Great job @eriksjolund! 🥇
You really organized the deployment of the whole thing (database and all services) and it looks like it's pretty much all done.

I've followed the deployment on a VM running Fedora 32 and looks like it's running fine, except for one step:
`loginctl enable-linger $USER` , that returns `access denied`.

<s>Another thing that I couldn't figure out is at what address/port scout is available from outside the VM (see my comment below).</s>This works,  but the pages are quite slow to load!

I havent't tested all the different types of deployment (USB, AWS, Ubuntu, etc) but looks very cool!

Overall I think the guide is quite easy to follow. There are some small things that would be nice to fix before merging:
- [ ] Could you update the changelog with a small description of the PR?
- [ ] for consistency I think the systemd folder should be placed at the same level of the kubernetes one, see structure here:https://github.com/Clinical-Genomics/scout/tree/master/containers/kubernetes. Could you update the howto txt also taking into account of the changes path of the .service files?
- [ ] We already have a [document](https://github.com/Clinical-Genomics/scout/blob/master/docs/admin-guide/container-deploy.md) in the admin guide describing how it works with Docker images and deploying with Kubernetes. You could either add your content to this file, perhaps creating a file content index, or you could add a link to another file, which I think should be included in the admin-guide folder and not in systemd (that should only contain the .service files).

Haven't tested the github action, because this PR is already big as it is, but we can have it fixed in another PR perhaps!
 
PS: it's fine because this is just documentation but perhaps next time you open a PR in scout you could do it from the official repo and not your forked one, so there will be no problems with tests and github actions not passing.

**Comment by northwestwitch** [comment]
Wait, it works outside the VM, it just super slow to load the pages:

image

File: systemd/README.md
Comment (eriksjolund) [reply systemd/README.md:73]: @northwestwitch I'll continue answering the other comment questions later. I'll start first with this question.
It works differently on different Linux distributions. On some Linux distributions `sudo loginctl enable-linger $USER` is needed.
On Ubuntu 20.04 it seems no `sudo` is needed.

This Github Actions workflow 
https://github.com/eriksjolund/scout/blob/systemd_services_podman_v3_github_actions/.github/workflows/test_systemd_user_service.yml
(that is running Ubuntu 20.04) use the command without the `sudo`.

**Comment by northwestwitch** [comment]
Hi @eriksjolund! We are planning a new scout release tomorrow and it would be nice to merge this one before. If you don't have time to work on these fixes I suggested in the review I could do it for you?

**Comment by northwestwitch** [comment]
@eriksjolund I've moved the unit service files under containers/systemd and I've created a new dedicated entry in the admin user guide having this structure:

image

**Review by northwestwitch — APPROVED** [review]

**Comment by northwestwitch** [comment]
Oh no, I can't merge this PR because it can't complete the coveralls step after the tests (it comes from Erik's forked repo and doesn't have the coveralls token)

image

Could you merge it as admin @moonso ? It's all good and it's only documentation anyway , thanks!

**Pull request merged**
